### PR TITLE
Avoid showing the builder while typing type args

### PR DESCRIPTION
**Pull request by rchande**
Fixes https://github.com/dotnet/roslyn/issues/14492

Tag @dotnet/roslyn-ide for review

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -894,6 +894,15 @@ private IEnumerable<TypeInferenceInfo> InferTypeInBinaryOrAssignmentExpression(E
                 var otherSideTypes = GetTypes(otherSide);
                 if (otherSideTypes.Any())
                 {
+                    // Don't infer delegate types except in assignments. They're unlikely to be what the
+                    // user needs and can cause lambda suggestion mode while
+                    // typing type arguments:
+                    // https://github.com/dotnet/roslyn/issues/14492
+                    if (!(binop is AssignmentExpressionSyntax))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Not sure how i feel about this.  Will need to think about this.

**Comment by rchande** [comment]
retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
Yup.  I'm ok with this now 👍    Felt strange at first.  But i think it's sensible.

**Comment by rchande** [comment]
**Ask Mode Info**
- **Customer scenario**: The builder appears while the user is trying to type type arguments, making it difficult to commit completion.
- **Bugs this fixes**: #14492
- **Workarounds**: None
- **Risk**: Low
- **Performance impact**: None.
- **Is this a regression from a previous update?**: No
- **How was the bug found?**: customer feedback

**Comment by srivatsn** [comment]
Approved pending one more review.

**Pull request merged**